src/scripts/domain/game.ts

```
@@ -51,15 +68,55 @@ export class Game implements GameInfo, GameInteractable {
 
+  /**
+   * Filter attack moves so that only the
+   * attack moves that take the most pieces
+   * are left
+   */
+  private filterAttackMoves() {
+    // Calculate max pieces taken
+    let maxPiecesTaken = 1;
+    for (const moveInfo of this._moveInfos) {
+      for (const move of moveInfo.moves) {
+        maxPiecesTaken = Math.max(maxPiecesTaken, move.toRemove!.length);
```

Biome (1.9.4)

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)